ghcide/exe/Main.hs

```diff
@@ -50,13 +50,18 @@ main = withTelemetryLogger $ \telemetryLogger -> do
     if argsVersion then ghcideVersion >>= putStrLn >> exitSuccess
     else hPutStrLn stderr {- see WARNING above -} =<< ghcideVersion
 
-    whenJust argsCwd IO.setCurrentDirectory
+    -- getHieDbLoc takes a directory path (the project root) and hashes it to find the location of the hiedb
+    -- when running commands directly from GHCIDE we need to provide the ABSOLUTE path to the project root (that's what HLS uses)
+    argsCwd <-case argsCwd of
+      Nothing   -> IO.getCurrentDirectory
+      Just root -> IO.setCurrentDirectory root >> IO.getCurrentDirectory
```

Comment:
This is a little obscure: why not just call `makeAbsolute` on `root`? This relies on you knowing that `getCurrentDirectory` returns an absolute path.

Reply:
You're right and I originally did it that way, but I wanted to match the semantics found throughout out the codebase. For example, in `Development.IDE.Main.defaultMain` all of the `argCommands` use the same `maybe IO.getCurrentDirectory return rootPath` expression to get the project root.